### PR TITLE
Contact shadows and volumetric to async

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
### Purpose of this PR
This PR adds an option to run contact shadows and volumetrics voxelization to async compute. 

---
### Release Notes

Added option to run Contact Shadows and Volumetrics Voxelization stage in Async Compute.

---
### Testing status
**Katana Tests**:  https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=contact-shadows-and-volumetric-to-async&automation-tools_branch=add-platform-filter&unity_branch=trunk [Just started] 

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**:  Medium

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 1]
```diff
@@ -381,23 +381,28 @@ void InitShadowSystem(HDRenderPipelineAsset hdAsset)
...
+        }
+
         void DeinitShadowSystem()
         {
             m_ShadowManager.Dispose();
             m_ShadowManager = null;
         }
 
-        int GetNumTileBigTileX(HDCamera hdCamera)
+        public int GetNumTileBigTileX(HDCamera hdCamera)
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: Question: Do we have a coding standard policy on the order of methods? i.e. All public go before the privates? Or is it not defined?

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: We are suppose to follow this convention: https://ono.unity3d.com/unity-extra/unity-meta/files/@/ReferenceSource/CSharp/Assets/CSharpReference.cs

**Comment by sebastienlagarde** [comment]
Katana master is red, need to wait until we fix it to have green katana

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs (hunk 1 above)
Comment (EvgeniiG) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: Don't see anything in the doc, but public before private is a reasonable strategy.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs [hunk 2]
```diff
@@ -185,6 +185,8 @@ static void Drawer_SectionAsyncComputeSettings(FrameSettingsUI s, SerializedFram
...
                 area.Add(p.runSSRAsync, SSRAsyncContent, () => p.overrideSSRInAsync, a => p.overrideSSRInAsync = a, () => p.enableAsyncCompute.boolValue, defaultValue: defaultFrameSettings.runSSRAsync, indent: 1);
                 area.Add(p.runSSAOAsync, SSAOAsyncContent, () => p.overrideSSAOInAsync, a => p.overrideSSAOInAsync = a, () => p.enableAsyncCompute.boolValue, defaultValue: defaultFrameSettings.runSSAOAsync, indent: 1);
+                area.Add(p.runContactShadowsAsync, contactShadowsAsyncContent, () => p.overrideContactShadowsInAsync, a => p.overrideContactShadowsInAsync = a, () => p.enableAsyncCompute.boolValue, defaultValue: defaultFrameSettings.runContactShadowsAsync, indent: 1);
+                area.Add(p.runVolumetricsVoxelizationAsync, volumetricsVoxelizationAsyncContent, () => p.overrideVolumetricsVoxelizationInAsync, a => p.overrideVolumetricsVoxelizationInAsync = a, () => p.enableAsyncCompute.boolValue, defaultValue: defaultFrameSettings.runVolumetricsVoxelizationAsync, indent: 1);
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs]: Better call it VolumeVoxelization, just like the pass.

**Comment by sebastienlagarde** [comment]
Fail test is an instability on vfx graph unrelated to this PR

**Pull request merged**